groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -2405,22 +2555,26 @@ class variant :
     variant() = default;
         // Create variant object containing the 0th alternative.  If variant is
         // allocator aware, use the currently installed default allocator to
-        // supply memory.  For simplicity of implementation, this method
-        // differs from the standard in the following :
+        // supply memory. For simplicity of implementation, this method differs
+        // from the standard in the following :
         //  - 'constexpr' is not implemented
         //  - 'noexcept' specification is not implemented
+        // This constructor is deleted if the 0th alternative is not default
+        // constructible.
```

Comment:
We can do that. I defaulted to deleting it because I'm familiar with gcc implementation and they have this bug. Constrained the default constructor and deleted the default constructor base.